src/gamma.jl

```diff
@@ -822,31 +799,38 @@ Compute the natural logarithm of the absolute value of the [`beta`](@ref) functi
 
 See also [`logbeta`](@ref).
 """
-function logabsbeta(a::Real, b::Real)
-    if a <= 0.0
-        if isinteger(a) && 1-a-b > 0
-            sgn = isinteger(b/2) ? 1 : -1
-            return logabsbeta(1-a-b,b)
-        end
+function logabsbeta(a::T, b::T) where T<:Real
+    # ensure that a <= b
+    if a > b
+        return logabsbeta(b, a)
     end
-    if b <= 0.0
-        if isinteger(b) && 1-a-b > 0
-            sgn = isinteger(a/2) ? 1 : -1
-            return logabsbeta(1-a-b,a)
+
+    if a <= 0.0 && isinteger(a)
+        if a + b <= 0
+            r = logbeta(1 - a - b, b)
+            sgn = isinteger(b/2) ? 1 : -1
+            return r, sgn
+        else
+            return -log(zero(first(promote(a, b)))), 1
         end
     end
-    if a < b
-        a,b = b,a
-    end
-    #asymptotic expansion for log(B(a,b)) for |a| >> |b|
-    if abs(a) > 1e5*abs(b) && abs(a) > 1e5
-        return (loggammadiv(b,a) + loggamma(b)), 1
+
+    # asymptotic expansion for large `b` and positive arguments
+    # FIXME! We probably lose precision for negative arguments. However, `loggammadiv`
+    # is based on the NSWC code which doesn't bother with negative arguments
+    if a > 0 && b > 8
```

Comment:
Are these thresholds based on the NSWC code?

Reply:
Yes. All the beta functions in NSWC assume positive arguments and the cutoff used there is `b>8`.